### PR TITLE
Update references to using `--find-links` for new index

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Fixes https://github.com/iree-org/iree/issues/10479

skip-ci

### CODE REVIEW

File: docs/website/docs/bindings/python.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ Stable release packages are published to
     run `pip install` with this extra option:
 
     ```
-    --find-links https://github.com/iree-org/iree/releases
+    --find-links https://iree-org.github.io/iree/pip-release-links.html
```
Comment (ScottTodd) [inline docs/website/docs/bindings/python.md:104]: `--find-links` has a short flag `-f`, which we use in many places too, like https://github.com/iree-org/iree/blob/65f8ce3f10422afe2a0b1368c10f4426478cfb78/docs/developers/debugging/tf_integrations_test_repro.md?plain=1#L13-L17

I can update https://github.com/iree-org/iree/tree/main/samples/colab and make a pass at some of the docs. Want to expand your search a bit in this PR, before calling the issue fixed?

Comment (GMNGeoffrey) [reply docs/website/docs/bindings/python.md:104]: Ahhh I was surprised at how few places I found. Yeah let me add those

Comment (ScottTodd) [reply docs/website/docs/bindings/python.md:104]: There will also be a long tail of uses outside of this repo to watch for, like https://github.com/iree-org/iree-jax/blob/28ae03dc4f734b96d64c70799d7d980fe1f2ffd6/build_tools/pip_install.py#L36

Worth a PSA on the email list too (once we've fixed the uses in this repo and have working examples to point people at)

Comment (GMNGeoffrey) [reply docs/website/docs/bindings/python.md:104]: Got everything except the colabs. Those need to be actually regenerated, I think?

Comment (ScottTodd) [reply docs/website/docs/bindings/python.md:104]: > Got everything except the colabs. Those need to be actually regenerated, I think?

You can technically edit the json like
```
      "source": [
        "!python -m pip install iree-compiler iree-runtime iree-tools-tf -f https://github.com/iree-org/iree/releases"
      ],
```

but it's worth going in and making those changes manually to verify that the notebooks still run and to get updated cell outputs.

Comment (GMNGeoffrey) [reply docs/website/docs/bindings/python.md:104]: Ok, you want to do that first, or should I drop the "fixes" from this?

Comment (ScottTodd) [reply docs/website/docs/bindings/python.md:104]: You can keep the "fixes" here.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**